feat: Add psychometric test evaluation

cbueth opened the pull request:
This PR adds the evaluation logic for the MultiplEYE psychometric tests as recorded with `MultiplEYE-psychometric-tests` (https://github.com/MultiplEYE-COST/MultiplEYE-psychometric-tests/), covering:
- Lewandowsky WMC battery (`MU`, `OS`, `SS`, `SSTM`)
- RAN task
- Stroop
- Flanker
- PLAB
- WikiVocab

It provides robust helpers for computing reaction time (RT) and accuracy, validates inputs, and parametrized unit test suite.

### What’s included
- New/updated processing functions in `preprocessing/psychometric_tests/preprocess_psychometric_tests.py`:
  - `preprocess_stroop(stroop_flanker_dir: Path) -> DataFrame`:
    - Returns RT/accuracy grouped by `stim_type` with columns `rt_mean`, `accuracy`, `num_items`.
  - `preprocess_flanker(stroop_flanker_dir: Path) -> DataFrame`:
    - Same grouped output by `stim_type`.
  - `preprocess_plab(plab_dir: Path) -> tuple[float, float, int]`:
    - Returns ungrouped `(rt_mean, accuracy, num_items)`.
  - `preprocess_ran(ran_dir: Path) -> DataFrame`:
    - Returns `Trial` and `Reading_Time` from CSV - audio time cut by lab.
  - `preprocess_wikivocab(wv_dir: Path) -> dict`:
    - Returns a dict with `rt_mean`, `accuracy`, `num_items`, and LexTALE‑style metrics: `num_sudo_words`, `num_real_words`, `sudo_correct`, `real_correct`, `overall_correct`, `incorrect_correct_score`.
  - `preprocess_lwmc(lwmc_dir: Path) -> dict`:
    - Computes per‑trial means and task means for `MU`, `OS`, `SS` from the unified CSV. Extracts `SSTM` from legacy `SSTM-<id>.dat` and divides by 240. Returns task scores/times and `LWMC_Total_score_mean`.
  - Shared helpers:
    - `_reaction_time_accuracy(...)` – shared grouped/ungrouped RT and accuracy calculation with `correct_only` option.
    - `__validate_rt_acc_inputs(...)` – input validation (NaN alignment, numeric RT, booleanish correctness).
    - `_find_one_filetype_with_columns(folder, columns, allow_nan=False)` – ensures exactly one CSV with required columns, optionally forbidding NaNs.

- New tests in `tests/psychometric_tests/test_preprocess_psychometric_tests.py`:
  - Parametrized tests for all six preprocessors (basic behavior plus error paths).
  - Direct tests for `_reaction_time_accuracy` and `_find_one_filetype_with_columns`.
  - Fixtures use a minimal `make_text_file` factory to generate tiny synthetic CSV/`.dat` inputs.

### Why
- Implements the evaluation step for psychometric tests.
- Ensures edge cases are handled (empty frames, NaN mismatches, non‑numeric RTs, invalid correctness tokens, malformed files, missing files, etc.).

### Output structure
There are two outputs venues when running `preprocess_all_participants`:
1. General overview table.
2. Participant output. CSV with all outputs, saved in the user folder.

1) Overview CSV (one row per participant) saved directly under the
   psychometric-tests-sessions folder. The filename is descriptive and
   includes the study/session tag (e.g. "SQ_CH_1_PT2"). The overview
   contains only the requested summary metrics:
   - LWMC: scores only (no times)
   - Stroop & Flanker: AccuracyEffect and TREffect only
   - WikiVocab: rt_mean, accuracy, incorrect_correct_score
   - RAN: Reaction time for two trials
   - PLAB: RT mean and accuracy

2) Per-participant detailed CSV placed in each participant folder with all
   available detailed metrics in a readable, wide format (namespaced
   columns). For example, grouped RT/accuracy for Stroop/Flanker are stored
   as columns like ``Stroop_congruent_rt_mean``.

### Validation
- NaN handling: NaN positions must match between RT and correctness to ensure paired statistics. Where no NaNs allowed at all: `allow_nan=False`.
- RT must be numeric. Correctness must be in `{0,1,True,False,NaN}`.
- Grouped outputs include `num_items` (non‑NaN RT counts per group) for context.
- `LWMC` computes per‑trial means to avoid overweighting longer trials. `SSTM` score read from `SSTM-<id>.dat`.

### How to test
- from `multipleye-preprocessing/`
- `pytest tests/psychometric_tests/test_preprocess_psychometric_tests.py`
- Furthermore:
  1. Set `DATASET_DIR` and `DATA_DIR` in `config.py`
  2. Run
  ```python
  if __name__ == '__main__':
       # Run full preprocessing
       preprocess_all_participants() 
   ```
   3. Check outputs in `psychometric-tests-sessions/` folder.

### File overview (touch points)
- `preprocessing/psychometric_tests/preprocess_psychometric_tests.py`: core implementations and helpers.
- `tests/psychometric_tests/test_preprocess_psychometric_tests.py`: comprehensive parametrized tests.
- `tests/conftest.py`: `make_text_file` fixture used by tests.

First round for #19, it will be closed in second feedback round with another PR.

cbueth committed:
add `pyproject.toml`

- uses pymovements `main` version
- remove `requirements.txt`

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
improve restructuring script

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
implement psychometric test preprocessing with tests

- added functionality to preprocess Stroop, Flanker, RAN, WikiVocab, and PLAB tests
- connected preprocessing pipeline with relevant config paths
- integrated tests to validate reaction time and accuracy computations

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
add detailed docstrings for psychometric test preprocessing functions

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
feat: specify Stroop and Flanker preprocessing to compute incongruent-congruent effects

- updated functions to return accuracy and reaction time differences between conditions
- added validation for required stimulus types
- revised docstrings for clarity and detailed effect definitions

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
specify sychometric test calculation and add saving

- expanded functions to compute and return detailed metrics for LWMC, RAN, Stroop, Flanker, WikiVocab, and PLAB
- revised Stroop and Flanker to output RT and accuracy grouped by stimulus type
- updated tests to validate metrics and expanded numeric output fields for preprocessing functions

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
fix: update LWMC scoring to use CSV data and improve computation logic

- replaced legacy `.dat` file scoring logic with CSV-based computations for MU, OS, and SS
- added mean reaction time output for each task
- retained SSTM scoring from `.dat` file
- updated returned dictionary with new key naming for consistency
- revised and expanded docstring for clarity on scoring methodology

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
update: test CSV outputs

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
expand: add tests for all psychometric preprocessors

- added validation tests for Stroop, Flanker, PLAB, RAN, WikiVocab, and LWMC preprocessing functions
- ensured test coverage for basic functionality and common error cases
- included mock input data to validate accuracy, reaction time, and error handling logic

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
Merge branch 'main' into feat/psychometric-tests

cbueth committed:
comment and remove `preprocess_all_participants_and_print` function

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
refactor: sudo -> pseudo

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
fix: change overview file name

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
refactor: use dicts for better readability and consistency

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
refactor: use 'sessions' instead of 'participants' in places better fitting

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
tests: update tests for changes from b1e0b8b6

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
add data_path_utils and tests

- chore: move psychometric tests into correct `unit/` folder

Signed-off-by: Carlson Büth <commit@cbueth.de>

theDebbister reviewed (approved):
thanks for the changes! It runs and looks good to me. As there will be another round of feedback with the entire team soon, we can merge this for now as a first version.

cbueth committed:
refactor: add second unit to column names for reaction times

- Renamed `*_Calculated` flags to `*_Done` across overview rows.
- Updated various metrics to use `_sec` suffix for time fields.

Signed-off-by: Carlson Büth <commit@cbueth.de>

cbueth committed:
Merge branch 'main' into feat/psychometric-tests
